src/test/architecture/test_architecture.py

```diff
@@ -312,3 +313,579 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
         caplog.clear()
         Architecture._verify([node_mock])
         assert len(caplog.record_tuples) == 1
+
+    # define original case of rename function
+    renaming_architecture_text = """
+named_paths:
+- path_name: target_path_0
+  node_chain:
+  - node_name: /node_0
+    publish_topic_name: /topic_0
+  - node_name: /node_1
+    subscribe_topic_name: /topic_0
+- path_name: target_path_1
+  node_chain:
+  - node_name: /node_1
+    publish_topic_name: /topic_1
+  - node_name: /node_0
+    subscribe_topic_name: /topic_1
+executors:
+- executor_type: single_threaded_executor
+  executor_name: executor_0
+  callback_group_names:
+  - /callback_group_0
+- executor_type: single_threaded_executor
+  executor_name: executor_1
+  callback_group_names:
+  - /callback_group_1
+nodes:
+- node_name: /node_0
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_1
+    callback_names:
+    - /callback_0
+    - /callback_1
+  callbacks:
+  - callback_name: /callback_0
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_1
+    callback_type: subscription_callback
+    topic_name: /topic_1
+    symbol: sub
+  publishes:
+  - topic_name: /topic_0
+    callback_names:
+    - /callback_0
+  subscribes:
+  - topic_name: /topic_1
+    callback_name: /callback_1
+- node_name: /node_1
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_0
+    callback_names:
+    - /callback_2
+    - /callback_3
+  callbacks:
+  - callback_name: /callback_2
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_3
+    callback_type: subscription_callback
+    topic_name: /topic_0
+    symbol: sub
+  publishes:
+  - topic_name: /topic_1
+    callback_names:
+    - /callback_2
+  subscribes:
+  - topic_name: /topic_0
+    callback_name: /callback_3"""
+
+    def test_rename_node(self, mocker):
+        # define case of node rename function
+        renamed_architecture_text = """
```

Comment:
I think the yaml string is redundant.
Consider using templates.

https://docs.python.org/ja/3/library/string.html#string.Template

Reply:
Fix it in [6beae0d](https://github.com/tier4/CARET_analyze/pull/156/commits/6beae0d15febd7991c450c6af9e2eb25d8d5bc99)